src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptionhandlers/ErrorNotificationExceptionHandler.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.exceptionhandlers;
+
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.IQueueClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.reform.bulkscanprocessor.client.ErrorNotificationException;
+
+import java.time.Instant;
+import javax.validation.constraints.NotNull;
+
+public class ErrorNotificationExceptionHandler {
+
+    private static final Logger log = LoggerFactory.getLogger(ErrorNotificationExceptionHandler.class);
+
+    private final IQueueClient errorNotificationPushClient;
+
+    public ErrorNotificationExceptionHandler(IQueueClient errorNotificationPushClient) {
+        this.errorNotificationPushClient = errorNotificationPushClient;
+    }
+
+    public IMessage handle(IMessage message, Throwable throwable) {
+        // return for logging part. and future requires something to be returned other than void
+        return throwable != null ? handleNonNullThrowable(message, throwable) : message;
+    }
+
+    private IMessage handleNonNullThrowable(IMessage message, @NotNull Throwable throwable) {
+        try {
+            throw (RuntimeException) throwable;
+        } catch (ClassCastException exception) {
+            log.error("Unable to cast Throwable to RuntimeException", throwable);
+
+            throw exception;
+        } catch (ErrorNotificationException exception) {
+            log.error("Failed to publish error notification", exception);
+
+            return handleErrorNotificationException(message, exception);
+        }
+    }
+
+    private IMessage handleErrorNotificationException(IMessage message, ErrorNotificationException exception) {
+        if (exception.getStatus().is5xxServerError()) {
+            // 10th delivery time is 6h delay
+            // Eventually it'll fail in case client is down for quite some time
+            long secondsToAdd = (long) Math.floor(Math.exp((double) message.getDeliveryCount()));
+
+            errorNotificationPushClient.scheduleMessageAsync(message, Instant.now().plusSeconds(secondsToAdd));
```

Comment:
While IMessageHandler doesn't offer any alternative, there are some problems with this approach, compared to complete/do nothing/dead-letter/abandon - we put a new message on the queue. What if we fail to do that? Also, would there be problems with duplicate detection? Not sure about delivery count being incremented in this case, either.

Reply:
now this is best intel - thanks. we have not configured queue to be [duplicate aware](https://github.com/hmcts/bulk-scan-shared-infrastructure/blob/b3836b8d89f233d018c7c60d0439047e8bb680a8/queue.tf#L20-L25) so this concern can be relaxed.

> what if we fail to do that?

well, message gets deadlettered disregarding

one thing is not clear - delivery count increase. aah! and i think i've understood the "gotcha" here. so i don't need to re-send "same" message to queue. i could just update scheduled time and throw the original exception. then `IMessageHandler` will understand it's a failure and return to the queue until delivery count has been reached. is this correct flow?